cmd/clusterctl/client/repository/components.go

```diff
@@ -205,8 +206,16 @@ func NewComponents(input ComponentsInput) (*components, error) {
 		return nil, err
 	}
 
+	if input.Options.ListVariablesOnly {
+		return &components{
+			variables:         variables,
+			targetNamespace:   input.Options.TargetNamespace,
+			watchingNamespace: input.Options.WatchingNamespace,
+		}, nil
+	}
+
 	processedYaml := input.RawYaml
-	if !input.Options.SkipVariables {
+	if !input.Options.SkipProcess {
```

Comment:
I will change this and return at this stage if we are skipping process, because all the subsequent steps assumes we are dealing with a yaml, which could not be true for processor different than the simpleProcessor

Reply:
Thinking more about this.
If we are returning at this stage, what is the difference between ListVariablesOnly and SkipProcess? should we merge those behaviors?